networkit/cpp/distance/Volume.cpp

```diff
@@ -51,21 +51,14 @@ std::vector<double> Volume::volume(const Graph &G, std::vector<double> rs, count
     double rmax = *std::max_element(std::begin(rs), std::end(rs));
     for (count j = 0; j < samples; j++) {
         std::unordered_map<node, double> ms = Volume::nodesWithinDistance(G, rmax, GraphTools::randomNode(G));
-        count i = 0;
-        for (const auto &r : rs) {
-            for (const auto &it : ms) {
-                if (it.second <= r) {
-                    xs[i] += 1;
-                }
-            }
-            i++;
-        }
+        for (count i = 0; i < rs.size(); ++i)
+            xs[i] += std::count_if(ms.begin(), ms.end(), [&](const auto &nodeDist) -> bool {
+                return nodeDist.second <= rs[i];
+            });
     }
     std::vector<double> ys;
-    ys.reserve(xs.size());
-    for (const auto &x : xs) {
-        ys.push_back(x / samples);
-    }
+    std::transform(xs.cbegin(), xs.cend(), std::back_inserter(ys),
+                   [samples](double x) { return x / static_cast<double>(samples); });
```

Comment:
Not sure if this is really simpler than the original code. Anyway, we should keep the `.reserve()`.